src/Parser.ts

```diff
@@ -807,7 +807,7 @@ export abstract class Parser extends Recognizer<Token, ParserATNSimulator> {
 			let ruleIndex: number = p.ruleIndex;
 			if (ruleIndex < 0) stack.push("n/a");
 			else stack.push(ruleNames[ruleIndex]);
-			p = p._parent;
+			p = p._parent as RuleContext;
```

Comment:
➡️ The issue is the use of a non-`undefined` default argument overrides the type annotation, so the compiler has assumed that `p` cannot be `undefined`. The solution here (a workaround assuming that this is a bug in the compiler) is assigning `p` to a local variable at the beginning of the method, and operating on the local variable instead of `p` itself.